interview-questions.md

```diff
@@ -9,41 +9,41 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: I know that `this` is used in a return and so it is used as refrencing method to identify objects in a function. 
 
-  Researched answer:
+  Researched answer: THIS is a direction to what is linked to, what it's attached to. THIS can be used to refer an object in a function. 
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a tool to help a developer see the work and all changes that has been made in real time, React also stores Javascript codes that is there to be used when needed by the developer. 
 
-  Researched answer:
+  Researched answer: React lets developers to have a fast, scalable, and simple experiance in front end framework that works with DOM (documet Object Model). 
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: render() method is required in a React class component because it can run the code without having to refresh the page.
 
-  Researched answer:
+  Researched answer: lifecycle method has 3 steps. mounting, updating, and unmounting. the mounting is important because it has the constructor, render, and methods. 
 
 
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
+  Your answer: if there is more than one items returned in JSX then there needs to be a fregment tag before and afte, while HTML since i dont have to return these fragments.
 
-  Researched answer:
+  Researched answer: in JSX, there must return a single parent element or the code wont compile. an example would be <>...</> and that also makes the code more readable. HTML is free and dont have to return a signle parent element. 
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: when yarn is ran, it is basically testing framework to check if the codebase is working or not. 
 
-  Researched answer:
+  Researched answer: yarn is javascript package manager and the package manager is to have code so when yarn start is ran, it opens a server and runs the code. 
```

Comment:
Yarn will manage all of the node modules by recognizing the dependancies in the package.json file.